packages/langium/src/validation/validation-registry.ts

```diff
@@ -106,8 +108,12 @@ export type ValidationChecks<T> = {
  *
  * `built-in` checks are errors produced by the lexer, the parser, or the linker. They cannot be used
  * for custom validation checks.
+ *
+ * You can also provide user-defined categories. These check will be skipped by default. Then it's up
+ * to you to schedule these checks: after the fast checks are done, or after saving a document,
+ * or with an explicit command, etc.
  */
-export type ValidationCategory = 'fast' | 'slow' | 'built-in'
+export type ValidationCategory = 'fast' | 'slow' | 'built-in' | string;
```

Comment:
This leads to `ValidationCategory` collapsing to just `string`, hence losing IntelliSense for the built-in ones. The work-around for this is to define it as such:
```suggestion
export type ValidationCategory = 'fast' | 'slow' | 'built-in' | (string & {});
```